mlflow/onnx.py

```diff
@@ -176,6 +188,7 @@ def save_model(
         onnx_version=onnx.__version__,
         data=model_data_subpath,
         providers=onnx_execution_providers,
+        onnx_session_options=onnx_session_options,
```

Comment:
we can add simple verification:
 - check keys are valid
 - for `execution_mode`, check value are valid
 - for other option keys, check types are correct, e.g., inter_op_num_threads should be integer that is > 0